### PR TITLE
fix: update stack deployment to follow Users&Roles best practices

### PR DESCRIPTION
**Pull request by mdelapenya**
- bump stack version 8.1.0-aa69d697
- fix: use new kibana roles and users
- feat: add a method for checking ES cluster health
- fix: create fleet-server after getting a service token from elasticsearch

## What does this PR do?
This PR tries to address the recent issue for the project:

1. it bumps the versions, cherry-picking #2052
1. it mimics what the APM ITs project is doing to load users and roles for elasticsearch. For that we are adding `admin` user as the super user, included in the `kibana_system` role.
1. it retrieves an Elasticsearch service token that will be used by fleet-server to start
1. it removes fleet-server from the stack profile, and manually adds it to the deployment with proper configuration, including the aforementioned service token

## Why is it important?
Two major things:
- tests with security (users-roles)
- bumps to latest versions
- fix fleet-server communication with kibana/elasticsearch

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have run the Unit tests (`make unit-test`), and they are passing locally
- [x] I have run the End-2-End tests for the suite I'm working on, and they are passing locally
- [ ] I have noticed new Go dependencies (run `make notice` in the proper directory)

## Related issues

- Bumps versions from #2052

### CODE REVIEW

**Comment by mdelapenya** [comment]
I'm skipping backport because of the cherry-picked commit including versions for 8.1

Will revert that commit once this is fixed, adding the labels for the proper backports

File: .ci/ansible/tasks/setup_test_script.yml [hunk 1]
```diff
@@ -1,4 +1,18 @@
 ---
+- name: Extend environment for Stack Bootstrapping
+  lineinfile:
+    state: present
+    line: "{{ itemd }}"
```
Comment (mdelapenya) [inline .ci/ansible/tasks/setup_test_script.yml:5]:
```suggestion
    line: "{{ item }}"
```

**Review by adam-stokes — APPROVED** [review]

**Comment by adam-stokes** [comment]
@mdelapenya looks like there is some sort of internal server error happening now

**Review by kuisathaverat — APPROVED** [review]

**Comment by mdelapenya** [comment]
Closer: it seems fleet-server port is not listening on 8220:

- SSHing into the stack machine and I check that is possible to telnet into localhost:8220
- SSHing into a worker, and I can check that is NOT possible to telnet into 3.138.33.117:8220 (the stack IP)

File: cli/config/compose/services/elastic-agent/fleet-server/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      - "FLEET_SERVER_SERVICE_TOKEN=${fleetServerServiceToken:-}"
+      - "FLEET_ENROLL=${fleetEnroll:-1}"
+      - "FLEET_ENROLLMENT_TOKEN=${fleetEnrollmentToken:-}"
+      - "FLEET_INSECURE=${fleetInsecure:-0}"
+      - "FLEET_URL=${fleetUrl:-}"
+      - "KIBANA_FLEET_HOST=http://kibana:5601"
+      - "KIBANA_FLEET_SETUP=${fleetServerMode:-0}"
+    platform: ${stackPlatform:-linux/amd64}
+    ports:
+      - "127.0.0.1:${fleetServerPort:-8220}:8220"
```
Comment (mdelapenya) [inline cli/config/compose/services/elastic-agent/fleet-server/docker-compose.yml:26]:
```suggestion
      - "${fleetServerPort:-8220}:8220"
```

File: internal/kibana/policies.go [hunk 3]
```diff
@@ -193,12 +193,12 @@ type DataStream struct {
...
 type Input struct {
-	Type           string        `json:"type"`
-	Enabled        bool          `json:"enabled"`
-	Streams        []interface{} `json:"streams"`
-	Vars           Vars          `json:"vars,omitempty"`
-	Config         interface{}   `json:"config,omitempty"`
-	CompiledStream interface{}   `json:"compiled_stream,omitempty"`
+	Type           string      `json:"type"`
+	Enabled        bool        `json:"enabled"`
+	Streams        []Stream    `json:"streams",omitempty`
```
Comment (kvch) [inline internal/kibana/policies.go:198]:
```suggestion
	Streams        []Stream    `json:"streams,omitempty"`
```

Comment (mdelapenya) [reply internal/kibana/policies.go:198]: Ah thanks for the fix and the recommendation. Will push the initialisation ASAP

**Comment by mdelapenya** [comment]
I'd say this PR is putting the code base into a state that satisfy latest changes in kibana and fleet. This is ready to merge on my side, although we are still seeing 30-40 test errors that are more or less consistent. I guess that once #2096 is addressed by @michel-laterman they will be solved.

@adam-stokes if you agree, I think we should merge this one.

**Comment by adam-stokes** [comment]
@mdelapenya +1 on merging

**Comment by mdelapenya** [comment]
Given this pipeline is almost green https://beats-ci.elastic.co/blue/organizations/jenkins/e2e-tests%2Fe2e-testing-mbp/detail/PR-2064/44/pipeline, we are going to merge this PR and open issues for the following items:

- APM Server scenarios: the agent never gets to the online status. I checked locally and it works, so my guesses are the code tries to bootstrap another stack because it uses the docker-compose provider for starting the agent container. Maybe using a vanilla testcontainers container request would help 🤔 
- k8s-autodiscover for elastic-agent: an issue already exists https://github.com/elastic/e2e-testing/issues/1992

**Comment by mdelapenya** [comment]
@jlind23 @ph we are in a much better situation to start with the support matrix once we merged this little baby. I'm gonna send a post-mortem with all the things we had to do (listed as commits in this PR)

Please relate to https://github.com/elastic/e2e-testing/pull/2064#issuecomment-1031819651 for the existing failures

**Pull request merged**